apps/ui/src/networks/starknet/index.ts

```diff
@@ -126,6 +128,22 @@ export function createStarknetNetwork(networkId: NetworkID): Network {
         }, 2000);
       });
     },
+    waitForIndexing: async (
+      txId: string,
+      timeout = 10000
```

Comment:
This is not enough in most cases, Starknet mainnet has 30s block time. API will likely have it accessible earlier (because of optimistic indexing), but we can't check it like this sadly. I don't know, maybe for now it might be better to ignore it for Starknet.

Reply:
I know that this works, but I'm not sure if it's huge improvement when we have basically 66.6% chance that it won't resolve successfully and will just fallback to 10s delay anyway.

Reply:
What are you proposing ? Increase the timeout for starknet ?

The improvement is mainly on the evm side, starknet is a bonus (it only costs a few requests to check transaction)